marlowe-playground-client/src/LoginPopup.purs

```diff
@@ -0,0 +1,103 @@
+module LoginPopup where
+
+import Prelude
+import Auth (AuthRole)
+import Control.Monad.Except (runExcept)
+import Data.Either (Either(..), hush)
+import Data.Maybe (Maybe(..))
+import Data.Traversable (for_)
+import Effect (Effect)
+import Effect.Aff (Aff, finally, makeAff, nonCanceler)
+import Effect.Class (liftEffect)
+import Effect.Ref (Ref)
+import Effect.Ref as Ref
+import Foreign.Generic.Class (encode, decode)
+import Web.Event.Event (EventType(..), Event)
+import Web.Event.EventTarget (EventListener, addEventListener, eventListener, removeEventListener)
+import Web.HTML as Web
+import Web.HTML.Location (replace)
+import Web.HTML.Window (outerHeight, outerWidth)
+import Web.HTML.Window as Window
+import Web.HTML.WindowExtra (close, postMessage)
+import Web.Socket.Event.MessageEvent as MessageEvent
+
+-- This function is intended to be called between a parent window. It creates a popup to the
+-- the Github auth page and waits for the popup to send a message with the current AuthRole
+openLoginPopup :: Aff AuthRole
+openLoginPopup = do
+  window <- liftEffect Web.window
+  let
+    popupHeight = 620
+
+    popupWidth = 600
+
+    githubLoginPage = "/api/oauth/github"
+
+    features :: Aff String
+    features =
+      liftEffect do
+        top <-
+          outerHeight window
+            <#> \windowHeight -> windowHeight / 2 - popupHeight / 2
```

Comment:
interesting, `<#>` hasn't really been used in this project so far but it's nice here, I may starting using it elsewhere.